### ISSUE TITLE
`CustomTextCLIP` vs `CLIP` training loss curves

### ISSUE THREAD

**Issue by yaoyu-33** [mlfoundations/open_clip#388]
I tried to compare training with  `CustomTextCLIP` class and default `CLIP` class. It seems their training loss curve looks slightly different. 

When I checked the source code of each module they are using, they are quite identical. So I cannot figure why their loss curve looks different. I was wondering what is the difference between `CustomTextCLIP` class and default `CLIP` class.

**Comment by usuyama** [comment]
Which model configs are you comparing?
OpenAI ViT is slightly different from timm version.

**Comment by yaoyu-33** [comment]
Hi, I am comparing these two classes. 
1. [default `CLIP` class](https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/model.py#L159)
2. [CustomTextCLIP](https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/model.py#L215)
I am training from scratch, not using any existing weights.

**Comment by rom1504** [comment]
they're the same

what config are you using ?

**Comment by yaoyu-33** [comment]
I am running with exactly the same config, I am not sure why they yield different loss. It's 2layer B/32 toy model and few thousand samples dummy data. To swap `CLIP` to `CustomTextCLIP`, I just swap their naming, i.e. `CLIP` => `CustomTextCLIP` and `CustomTextCLIP` => `CLIP`. I am very confused to see slightly different loss curve.
My full command looks like
```
torchrun --nproc_per_node 1 -m training.main  \
  --train-data '/???/00328-00332_{000000..000007}.tar' \
  --train-num-samples 8000 \
  --val-data '/???/00328-00332_000008.tar' \
  --dataset-type webdataset  \
  --batch-size 200  \
  --precision fp32  \
  --workers 4 \
  --model ViT-B-16_2layer \
  --lr 5e-4 \
  --report-to wandb \
  --dataset-resampled
```
<img width="419" alt="image" src="https://user-images.githubusercontent.com/54727607/215294211-e16eb315-6acd-428e-8c8a-e267fcfa8be6.png">

**Comment by rwightman** [comment]
@yaoyu-33 have you done more that one run w/ each across different seeds?

**Comment by yaoyu-33** [comment]
Hi @rwightman , I didn't change the seed, so it always 0.
I run more than once, each time the loss curve looks similar with if I am using the same Model. I will try to add few more runs with different seed and show here.

**Comment by yaoyu-33** [comment]
Thanks for the help. Close this issue. It seems to come from initialization differences.
